src/Resolver.js

```diff
@@ -217,19 +217,29 @@ class Resolver {
     return pkg;
   }
 
-  getPackageMain(pkg) {
+  async getPackageMain(pkg) {
     // libraries like d3.js specifies node.js specific files in the "main" which breaks the build
     // we use the "module" or "jsnext:main" field to get the full dependency tree if available
-    let main = [pkg.module, pkg['jsnext:main'], pkg.browser, pkg.main].find(
-      entry => typeof entry === 'string'
-    );
+    const entries = [pkg.module, pkg['jsnext:main'], pkg.browser, pkg.main]
+      .filter(entry => typeof entry === 'string')
+      .map(entry => path.resolve(pkg.pkgdir, entry));
+
+    // main is the first entry that exists
+    for (let i = 0; i < entries.length; ++i) {
+      const entry = entries[i];
 
-    // Default to index file if no main field find
-    if (!main || main === '.' || main === './') {
-      main = 'index';
+      try {
+        const stat = await fs.stat(entry);
+        if (stat.isDirectory()) return path.resolve(entry, 'index');
+        return entry;
+      } catch (e) {
```

Comment:
I think we should just blow-up here if an entry record declared in package.json doesn't exists instead of handling it.